### PR TITLE
[G2M] Widget - create barline widget

### LINKED ISSUE #206: Bar-Line - implement controls to configure a Bar-Line chart widget type
**Implementation:**
- add configuration controls for a Bar-Line chart widget
Screen Shot 2022-12-09 at 2 00 56 PM

### PR DESCRIPTION
**Pull request by geoerika**
Closes #206

**Changes:**
1. Updated widget type icons
2. Added BarLine Widget controls

Screen Shot 2022-12-20 at 12 48 17 PM

**BarLine Widget config recording:**

https://user-images.githubusercontent.com/41120953/208733451-dc9e69a0-a1de-4e7d-8171-316c19c35d28.mov

3. Added Line Marker & Line Fill controls to Line Widgets

Screen Shot 2022-12-20 at 12 54 06 PM

4. Arranged switch controls on the right side in a grid instead of using flexbox & now the elements align better in the right side panel.

### CODE REVIEW

File: src/controls/editor-mode/components/color-scheme-controls.js [hunk 1]
```diff
@@ -118,32 +123,45 @@ const ColorSchemeControls = () => {
...
-    genericOptions: { baseColor: colord(v).toHex() },
+    genericOptions: { baseColor: { [selectedColor]: colord(v).toHex() } },
   }), 100)
 
   useEffect(() => {
-    update({ presetColors: presetColors.map((_c, i) => i === selectedColorIndex ? baseColor : _c) })
+    update({ presetColors: presetColors.map((_c, i) => i === selectedColorIndex ? color : _c) })
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [baseColor, selectedColorIndex, update])
+  }, [color, selectedColorIndex, update])
```
Comment (woozyking) [inline src/controls/editor-mode/components/color-scheme-controls.js:154]: The lint suppression comment before this one isn't needed anymore it seems

File: src/controls/editor-mode/components/unique-option-controls.js [hunk 2]
```diff
@@ -1,33 +1,59 @@
...
-                  i
-                )}
-              </div>
-            )
-        }
-      })
+    <div className={classes.uniqueOptionsContainer}>
+      {Object.entries(typeInfo[type]?.uniqueOptions || {})
+        .map(([k, { name, type: _type }], i) => {
+          switch (_type) {
```
Comment (woozyking) [inline src/controls/editor-mode/components/unique-option-controls.js:40]: perhaps add a `default` case and return `null`?

**Review by woozyking — APPROVED** [review]
looks good, with minor comments

**Review by Hamaad102 — APPROVED** [review]

File: src/controls/editor-mode/components/color-scheme-controls.js (hunk 1 above)
Comment (geoerika) [reply src/controls/editor-mode/components/color-scheme-controls.js:154]: Screen Shot 2022-12-20 at 12 54 06 PM
@woozyking it's still needed. For some reason Kyle decided not to include `presetColors` in the dep list. I will look at this later and keep it as it is.

**Review by vlatawiec — APPROVED** [review]
Looks good! Just noticed this now (not urgent) but ideally the label in the widget type selection would be "Bar-Line'. Otherwise labels and progress so far look good from ux side

File: src/controls/editor-mode/components/color-scheme-controls.js (hunk 1 above)
Comment (woozyking) [reply src/controls/editor-mode/components/color-scheme-controls.js:154]: ah I see, fair enough

**Pull request merged**